Shared.NFe.Utils/Assinatura/Assinador.cs

```diff
@@ -106,10 +116,11 @@ public static class Assinador
             }
             finally
             {
-                //Se não mantém os dados do certificado em cache e o certificado não foi passado por parâmetro(isto é, ele foi criado dentro deste método), 
```

Comment:
O atributo certificadoDigital nunca vai ser ser nulo, porque se ele for os atributos passados por parâmetro não conseguem criar o objeto.

E se o certificado não for nulo e der Reset() vai quebrar para quem controla o certificado por conta própria, visto que serviços como cancelamento chamam este método para assinar o serviço.